scripts/fs.py

```diff
@@ -76,51 +94,57 @@ def main(self):
                 logging.info('Mask the t1')
                 t1 = tmpdir / 't1masked.nii.gz'
                 ImageMath('3', t1, 'm', self.t1, self.t1mask)
-                skullstrip = '-noskullstrip'
 
             else:
-                skullstrip = '-skullstrip'
                 t1 = tmpdir / 't1.nii.gz'
                 self.t1.copy(t1)
             
+            subjid = self.t1.stem
+            common_params=['-s', subjid]
+
+
+            autorecon3_params=[]
             if self.t2:
                 if self.t2mask:
                     logging.info('Mask the t2')
                     t2 = tmpdir / 't2masked.nii.gz'
                     ImageMath('3', t2, 'm', self.t2, self.t2mask)
-                    skullstrip = '-noskullstrip'
-
+            
                 else:
-                    skullstrip = '-skullstrip'
                     t2 = tmpdir / 't2.nii.gz'
-                    self.t2.copy(t2)
+                    self.t2.copy(t2)  
+                
+                autorecon3_params= ['-T2', t2, '-T2pial']
+            
 
-            logging.info("Run freesurfer on " + t1)
-            subjid = t1.stem
+            logging.info("Running freesurfer on " + t1)
 
             if self.ncpu=='-1':
-                self.ncpu= N_CPU
+                self.ncpu= str(N_CPU)
 
             if int(self.ncpu)>1:
+                common_params+=['-parallel', '-openmp', self.ncpu]
 
-                if self.t2:
-                    bash['-c', 'recon-all -s '+subjid+' -i '+t1+' -T2 ' +t2+' -autorecon1 ' +skullstrip+' -parallel -openmp '+self.ncpu] & FG
-                else:
-                    bash['-c', 'recon-all -s '+subjid+ ' -i '+t1+' -autorecon1 '+skullstrip+' -parallel -openmp '+self.ncpu] & FG
-                (tmpdir / subjid / 'mri/T1.mgz').copy(tmpdir / subjid / 'mri/brainmask.mgz')
-                bash['-c', 'recon-all -autorecon2 -subjid '+subjid+' -parallel -openmp '+self.ncpu] & FG
-                bash['-c', 'recon-all -autorecon3 -subjid '+subjid+' -parallel -openmp '+self.ncpu] & FG
 
-            else: # intentionally writing a separate block omitting parallelization attributes
-                if self.t2:
-                    bash['-c', 'recon-all -s '+subjid+' -i '+t1+' -T2 ' +t2+' -autorecon1 ' +skullstrip] & FG
-                else:
-                    bash['-c', 'recon-all -s '+subjid+ ' -i '+t1+' -autorecon1 '+skullstrip] & FG
+            autorecon1_params=[]
+            if not self.no_hires:
+                common_params.append('-hires')
+                autorecon1_params=['-expert', self.expert_file]
+
+
+            # run recon_all in three steps so we can overwrite/provide MABS masked T1
+            # -noskullstrip is used with -autorecon1 only, so we need to check whether T1 is masked/T1 mask is provided
+            # irrespective of masked/unmasked T2
```

Comment:
See the rationale behind disregarding `--t2mask` for deciding `-skullstrip` vs `-noskullstrip`. The former is enabled by default.